src/components/SearchModal/sorting.ts

```diff
@@ -37,14 +48,20 @@ function getTokenComparator(balances: {
     }
   }
 }
-
-export function useTokenComparator(inverted: boolean): (tokenA: Token, tokenB: Token) => number {
+const EMPTY_ARRAY: any = []
+export function useTokenComparator(inverted: boolean, containsETH = false): (tokenA: Token, tokenB: Token) => number {
   const balances = useAllTokenBalances()
+  const { account, chainId } = useActiveWeb3React()
+  const accounts = useMemo(() => (containsETH ? [account] : EMPTY_ARRAY), [containsETH, account])
+  const ethBalance = useETHBalances(accounts)
+  const value = Object.values(ethBalance)[0]
+  // eslint-disable-next-line
+  const memoEthBalance = useMemo(() => value, [value?.toExact()]) // do not put value dependency here
```

Comment:
```
const memoEthBalance = useMemo(() => {
    return ethBalance[account]?.toExact()
},[ethBalance, account])
```

Reply:
I used to do like your comment,   but
when ethBalance1 !== ethBalance2 but same toExact value, I dont need it re-render

Reply:
re-render is not bad. if value dont change, it doesnt trigger real DOM update. so it's okay. 
It will be more clear